### PR TITLE
Petsc option database, and Picard iteration using Eigen executioner 

### PR DESCRIPTION
**Pull request by fdkong**
(1) Support PETSc option database  #13623

(2) Support Picard iteration in eigen executioner    #12767. Eigen executioner can serve as a  master or a sub.

### CODE REVIEW

**Comment by fdkong** [comment]
I need to get this in

File: framework/include/systems/NonlinearEigenSystem.h [hunk 1]
```diff
@@ -54,6 +54,16 @@ class NonlinearEigenSystem : public NonlinearSystemBase
 
   virtual NumericVector<Number> & RHS() override;
 
+  /*
+   *  A residual vector at MOOSE side for AX
+   */
+  NumericVector<Number> & ResidualVectorAX();
```
Comment (lindsayad) [inline framework/include/systems/NonlinearEigenSystem.h]: This method name should start lower case to distinguish it from a class and to be consistent with our style

File: framework/include/systems/NonlinearEigenSystem.h [hunk 2]
```diff
@@ -54,6 +54,16 @@ class NonlinearEigenSystem : public NonlinearSystemBase
...
+
+  /*
+   * A residual vector at MOOSE side for BX
+   */
+  NumericVector<Number> & ResidualVectorBX();
```
Comment (lindsayad) [inline framework/include/systems/NonlinearEigenSystem.h]: Same here

**Review by lindsayad — CHANGES_REQUESTED** [review]
Looks good other than the small suggestions

**Review by lindsayad — APPROVED** [review]

**Pull request merged**